shell/common/animator.cc

```diff
@@ -62,6 +62,10 @@ void Animator::BeginFrame(
     std::unique_ptr<FrameTimingsRecorder> frame_timings_recorder) {
   TRACE_EVENT_ASYNC_END0("flutter", "Frame Request Pending",
                          frame_request_number_);
+  // Clear layer trees rendered out of a frame. Only Animator::Render called
+  // within a frame is used.
+  layer_trees_tasks_.clear();
```

Comment:
Is this intentional? I thought the latest plan was to not clear layer trees at the start of the frame and allow the frame to overwrite a view's layer tree?

Reply:
The plan is to clear the layer trees at the frame start, so that out-of-vsync `Render`s are ignored, while multiple `Render`s for the same view ID are allowed with the latest one kept.

Reply:
(I should probably add a test for this.)

Reply:
Ok gotcha. Agreed a test sounds good. I'll mark this comment as resolved